src/crepe/facade/SDLContext.cpp

```diff
@@ -104,33 +108,34 @@ SDL_Rect SDLContext::get_src_rect(const Sprite & sprite) const {
 		.h = sprite.sprite_rect.h,
 	};
 }
-SDL_Rect SDLContext::get_dst_rect(const Sprite & sprite, const vec2 & pos,
-								  const double & scale, const Camera & cam) const {
+SDL_Rect SDLContext::get_dst_rect(const Sprite & sprite, const vec2 & pos, const Camera & cam,
+								  const double & img_scale) const {
+
+	int width = sprite.height * sprite.aspect_ratio;
+	int height = sprite.height;
 
-	double adjusted_w = sprite.sprite_rect.w * scale * cam.zoom;
-	double adjusted_h = sprite.sprite_rect.h * scale * cam.zoom;
-	double adjusted_x = (pos.x - cam.x) * cam.zoom - adjusted_w / 2;
-	double adjusted_y = (pos.y - cam.y) * cam.zoom - adjusted_h / 2;
+	width *= img_scale * cam.zoom;
+	height *= img_scale * cam.zoom;
 
 	return SDL_Rect{
-		.x = static_cast<int>(adjusted_x),
-		.y = static_cast<int>(adjusted_y),
-		.w = static_cast<int>(adjusted_w),
-		.h = static_cast<int>(adjusted_h),
+		.x = static_cast<int>((pos.x - cam.pos.x + (cam.viewport.x / 2) - width / 2)),
+		.y = static_cast<int>((pos.y - cam.pos.y + (cam.viewport.y / 2) - height / 2)),
+		.w = width,
+		.h = height,
 	};
 }
 
 void SDLContext::draw_particle(const Sprite & sprite, const vec2 & pos, const double & angle,
-							   const double & scale, const Camera & camera) {
+							   const double & img_scale, const Camera & cam) {
```

Comment:
Kun je uitleggen waarom er 2 draws zijn ik zie namelijk 2 keer exact dezelfde functie. 
Als je in de rendersystem het aanpast dat bij de normale draw ook de data doorstuurt ipv de transform kan 1 functie weg. Wat je ook kan doen is een struct maken is je render system die al deze data bevat (sprite,pos,angle,img_scale,camera) die meestuurt naar 1 draw functie dan ben je klaar.